tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp

```diff
@@ -354,6 +354,17 @@ std::vector<Shape> OptimizedConvNew::compute_output_shapes(const std::vector<Ten
     return {output_tensor_shape};
 }
 
+inline CoreRangeSet num_cores_to_corerange_set_block(uint32_t target_num_cores, CoreCoord grid_size, uint32_t num_cores_along_width = 0) {
+    uint32_t num_cores_x = grid_size.x;
+    uint32_t num_cores_y = grid_size.y;
+    uint32_t total_available_cores = num_cores_x * num_cores_y;
+    num_cores_along_width = num_cores_along_width > 0 ? num_cores_along_width : num_cores_x;
+    TT_FATAL(target_num_cores % num_cores_along_width == 0, "The grid is not a rectangle: target_num_cores = {}, num_cores_along_width = {}", target_num_cores, num_cores_along_width);
+    TT_FATAL(target_num_cores <= total_available_cores, "Target number of cores {} is greater than total number of available cores {}", target_num_cores, total_available_cores);
+    CoreRange core_range = {{0, 0}, {target_num_cores / num_cores_along_width - 1, num_cores_along_width - 1}};
+    return CoreRangeSet({core_range});
+}
+
```

Comment:
What is the point of this function?

Reply:
The generic version does not calculate the grid for block sharding correctly (with x != grid width)